### ISSUE TITLE
wails v2.8.2  Build error - could not determine kind of name for C.free

### ISSUE THREAD

**Issue by fanspace** [wailsapp/wails#3538]
### Description

**wails build  occur error**
Compiling application: # github.com/wailsapp/wails/v2/pkg/assetserver/webview
../../go/pkg/mod/github.com/wailsapp/wails/v2@v2.8.2/pkg/assetserver/webview/responsewriter_linux.go:113:2: could not determine kind of name for C.free

### To Reproduce

When running either wails dev or wails build, the same error occurs 

### Expected behaviour

finish the compilation

### Screenshots

![1718465106994](https://github.com/wailsapp/wails/assets/14883295/60b1c272-f12a-4694-8c21-b0dcf5e587ae)
![1718465119906](https://github.com/wailsapp/wails/assets/14883295/f84c7f0e-b86b-42b4-a6be-53614a70a7f3)

### System Details

```shell
# Wails
Version         | v2.8.2
Package Manager | apt   

# System
┌──────────────────────────────────────────────────────────┐
| OS           | Ubuntu                                    |
| Version      | 16.04                                     |
| ID           | ubuntu                                    |
| Go Version   | go1.22.4                                  |
| Platform     | linux                                     |
| Architecture | amd64                                     |
| CPU          | Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz  |
| GPU          | SVGA II Adapter (VMware) - Driver: vmwgfx |
| Memory       | 8GB                                       |
└──────────────────────────────────────────────────────────┘

# Dependencies
┌───────────────────────────────────────────────────────────────────────────┐
| Dependency | Package Name          | Status    | Version                  |
| *docker    | docker.io             | Available | 18.09.7-0ubuntu1~16.04.7 |
| gcc        | build-essential       | Installed | 12.1ubuntu2              |
| libgtk-3   | libgtk-3-dev          | Installed | 3.18.9-1ubuntu3.3        |
| libwebkit  | libwebkit2gtk-4.0-dev | Installed | 2.20.5-0ubuntu0.16.04.1  |
| npm        | npm                   | Installed | 7.10.0                   |
| *nsis      | nsis                  | Available | 2.50-1                   |
| pkg-config | pkg-config            | Installed | 0.29.1-0ubuntu1          |
└───────────────────────── * - Optional Dependency ─────────────────────────┘
```

### Additional context
Both on Ubuntu 18 and Ubuntu 20, Wails operates smoothly. However, due to various constraints with my client's computer which prevents an upgrade, it necessitates the use of glibc 2.23; hence, I am compelled to utilize an older version of Ubuntu as the compilation machine.
_No response_

**Comment by leaanthony** [comment]
The go version in your CLI screenshot is 1.20 but 1.22 in doctor....

**Comment by fanspace** [comment]
> The go version in your CLI screenshot is 1.20 but 1.22 in doctor....

Yes, initially, I had installed Go version 1.22 and encountered this issue. Suspecting that the problem might be due to the newer version of Go, I downgraded to version 1.20. However, I found that this action did not resolve the issue.

**Comment by fanspace** [comment]
> The go version in your CLI screenshot is 1.20 but 1.22 in doctor....

Due to various constraints, the client's computer is stuck with Ubuntu 16.04, unable to be upgraded or altered. Consequently, I've been attempting to package the application using Ubuntu 18.04 and then run it on 16.04, encountering issues along the way related to glibc version discrepancies and errors such as 'undefined symbol: gdk_display_get_monitor_at_window' with libgtk. I'm quite curious about the feasibility of running Wails version 2 on Ubuntu 16.04 and would greatly appreciate your guidance on how to proceed. Thank you very much.

**Comment by leaanthony** [comment]
Closed in error.

**Comment by leaanthony** [comment]
Closing as stale — this appears to be tied to a very old toolchain (Ubuntu 16.04 / glibc 2.23) which is well outside any supported range. No resolution was found and there's been no follow-up. If you're hitting a CGO issue on a current distro, please open a fresh issue with `wails doctor` output. Thanks!
